packages/cli/test/unit/commands/integration/add.test.ts

```diff
@@ -238,6 +238,31 @@ describe('integration', () => {
           await expect(exitCodePromise).resolves.toEqual(0);
           expect(openMock).not.toHaveBeenCalled();
         });
+
+        it('should require the vercel dashboard for expressions in UI wizard', async () => {
+          useProject({
+            ...defaultProject,
+            id: 'vercel-integration-add',
+            name: 'vercel-integration-add',
+          });
+          const cwd = setupUnitFixture('vercel-integration-add');
+          client.cwd = cwd;
+          client.setArgv('integration', 'add', 'acme-unsupported');
+          const exitCodePromise = integrationCommand(client);
+          await expect(client.stderr).toOutput(
+            `Installing Acme Product by Acme Integration under ${team.slug}`
+          );
+          await expect(client.stderr).toOutput(
+            'Do you want to link this resource to the current project? (Y/n)'
+          );
+          client.stdin.write('n\n');
+          await expect(client.stderr).toOutput(
+            'This resource must be provisioned through the Web UI. Open Vercel Dashboard?'
+          );
+          client.stdin.write('n\n');
+          await expect(exitCodePromise).resolves.toEqual(0);
+          expect(openMock).not.toHaveBeenCalled();
```

Comment:
Should there be a test case where the browser is opened?

Reply:
Technically speaking, yes, or this one should be the one where it's opened, and 'no' should be assumed to be covered in a different test case. I wrote this under the assumption the general "not supported" flow was already covered.